src/userspace/exec_monitor.cpp

```diff
@@ -65,11 +66,31 @@ int ExecMonitor::run()
 		std::cerr << "[EXEC_MONITOR]: Error allocating the ringbufffer.\n";
 		goto out;
 	}
-	
+
+	if (n_proc > 0) {
+		skel->bss->n_monitored_proc = 0;
+		skel->bss->max_n_proc = n_proc;
+	}
+
+	if (ppid_list.size() != 0) {
+		skel->bss->filter_by_ppid = 1;
+		int list_size = 10;
```

Comment:
10 feels like an arbitrary number. If this is tied to the size of an array or something, the way to do that is to use a #define statement where you give this a name. For example PPID_LIST_MAP_LEN and ideally define it in a header that is shared between userspace and kernelspace, so that one only has to change this constant and the behaviours of both the userspace and kernelspace will evolve all at once.